addons/payment_authorize/controllers/main.py

```diff
@@ -1,85 +1,58 @@
-        # Authorize.Net is expecting a response to the POST sent by their server.
-        # This response is in the form of a URL that Authorize.Net will pass on to the
-        # client's browser to redirect them to the desired location need javascript.
-        return request.render('payment_authorize.payment_authorize_redirect', {
-            'return_url': urls.url_join(base_url, "/payment/process")
-        })
-
-    @http.route(['/payment/authorize/s2s/create_json_3ds'], type='json', auth='public', csrf=False)
-    def authorize_s2s_create_json_3ds(self, verify_validity=False, **kwargs):
-        token = False
-        acquirer = request.env['payment.acquirer'].browse(int(kwargs.get('acquirer_id')))
-
-        try:
-            if not kwargs.get('partner_id'):
-                kwargs = dict(kwargs, partner_id=request.env.user.partner_id.id)
-            token = acquirer.s2s_process(kwargs)
-        except ValidationError as e:
-            message = e.args[0]
-            if isinstance(message, dict) and 'missing_fields' in message:
-                if request.env.user._is_public():
-                    message = _("Please sign in to complete the payment.")
-                    # update message if portal mode = b2b
-                    if request.env['ir.config_parameter'].sudo().get_param('auth_signup.allow_uninvited', 'False').lower() == 'false':
-                        message += _(" If you don't have any account, ask your salesperson to grant you a portal access. ")
-                else:
-                    msg = _("The transaction cannot be processed because some contact details are missing or invalid: ")
-                    message = msg + ', '.join(message['missing_fields']) + '. '
-                    message += _("Please complete your profile. ")
-
-            return {
-                'error': message
-            }
-
-        if not token:
-            res = {
-                'result': False,
-            }
-            return res
 
-        res = {
-            'result': True,
-            'id': token.id,
-            'short_name': token.short_name,
-            '3d_secure': False,
-            'verified': True, #Authorize.net does a transaction type of Authorization Only
-                              #As Authorize.net already verify this card, we do not verify this card again.
+    @http.route('/payment/authorize/get_acquirer_info', type='json', auth='public')
+    def authorize_get_acquirer_info(self, acquirer_id):
+        """ Return information on the acquirer.
+
+        :param int acquirer_id: The acquirer handling the transaction, as a `payment.acquirer` id
+        :return: Information on the acquirer, namely the state, login ID and client key
+        :rtype: dict
+        """
+        acquirer_sudo = request.env['payment.acquirer'].sudo().browse(acquirer_id).exists()
+        return {
+            'state': acquirer_sudo.state,
+            'login_id': acquirer_sudo.authorize_login,
```

Comment:
this field has a `groups='base.group_system'` and yet, we can retrieve it here

Reply:
I understand the confusion but this is intended behavior. The login ID is public and this is the recommended route response by Authorize.Net.
To make it clearer, we could eventually drop the `groups` attribute on the field but it was put there as an added security in case of something breaking/unprotected route writing on the field/pick your poison.

Reply:
Then it would be useful to add a comment here to remind that both the `authorize_login` and the `authorize_client_key` are public and can be shared - because they cannot be used to initiate a transaction, which requires the `transaction_key`.

Also, the `authorize_client_key` might have been named `authorize_public_client_key` or `authorize_public_key` to make that obvious without needing comments ;-)